packages/flutter/lib/src/material/mergeable_material.dart

```diff
@@ -0,0 +1,507 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+/// A [Widget] that abstracts over a Material slice and a Material gap.
+abstract class MergeableMaterialItem extends StatelessWidget {
```

Comment:
Because it can have a key. I'm using `Widget.canUpdate` on `MergeableMaterialItem`.